code/modules/scrap/oldificator.dm

```diff
@@ -0,0 +1,430 @@
+			R.volume = rand(0, R.volume)
+		reagents.add_reagent("toxin", rand(0, actual_volume - reagents.total_volume))
+
+
+//Sealed survival food, always edible
+/obj/item/weapon/reagent_containers/food/snacks/liquidfood/make_old()
+	return
+
+/obj/item/ammo_magazine/make_old()
+	var/del_count = rand(0,contents.len)
+	for(var/i = 1 to del_count)
+		var/removed_item = pick(stored_ammo)
+		stored_ammo -= removed_item
+		qdel(removed_item)
+	..()
+
+/obj/item/weapon/cell/make_old()
+	.=..()
+	if (.)
+		charge = min(charge, rand_between(0, maxcharge))
+		if(prob(10))
+			rigged = TRUE
+			if(prob(80))
+				charge = maxcharge  //make it BOOM hard
+
+/obj/item/weapon/stock_parts/make_old()
+	.=..()
+	if (.)
+		var/degrade = pick(0,1,1,1,2)
+		rating = max(rating - degrade, 1)
+
+
+/obj/item/stack/material/make_old()
+	return
+
+/obj/item/stack/rods/make_old()
+	return
+
+/obj/item/weapon/grenade/make_old()
+	..()
+	det_time = rand_between(0, det_time)
+
+/obj/item/weapon/tank/make_old()
+	.=..()
+	if (.)
+		air_contents.remove(pick(0.2, 0.4 ,0.6, 0.8))
+
+
+/obj/item/weapon/circuitboard/make_old()
+	.=..()
+	if (.)
+		if(prob(75))
+			name = T_BOARD("unknown")
+			build_path = pick(/obj/machinery/washing_machine, /obj/machinery/broken, /obj/machinery/shower, /obj/machinery/holoposter, /obj/machinery/holosign)
+
+
+/obj/item/weapon/aiModule/make_old()
+	.=..()
+	if (.)
+		if(prob(75) && !istype(src, /obj/item/weapon/aiModule/broken))
+			var/obj/item/weapon/aiModule/brokenmodule = new /obj/item/weapon/aiModule/broken
+			brokenmodule.name = src.name
+			brokenmodule.desc = src.desc
+			brokenmodule.make_old()
+			qdel(src)
+
+
+/obj/item/clothing/suit/space/make_old()
+	.=..()
+	if (.)
+		if(prob(50))
+			create_breaches(pick(BRUTE, BURN), rand(10, 50))
+
+
+/obj/item/clothing/make_old()
+	.=..()
+	if (.)
+		if(prob(30))
+			slowdown += pick(0.5, 0.5, 1, 1.5)
+		if(prob(50))
+			armor["melee"] = rand(0, armor["melee"])
+			armor["bullet"] = rand(0, armor["bullet"])
+			armor["laser"] = rand(0, armor["laser"])
+			armor["energy"] = rand(0, armor["energy"])
+			armor["bomb"] = rand(0, armor["bomb"])
+			armor["bio"] = rand(0, armor["bio"])
+			armor["rad"] = rand(0, armor["rad"])
+		if(prob(50))
+			heat_protection = rand(0, round(heat_protection * 0.5))
+		if(prob(50))
+			cold_protection = rand(0, round(cold_protection * 0.5))
+		if(prob(50))
+			contaminate()
+		if(prob(60))
+			add_blood()
+		if(prob(60)) // I mean, the thing is ew gross.
+			equip_delay += rand(0, 6 SECONDS)
+
+/obj/item/weapon/aiModule/broken
+	name = "\improper broken core AI module"
+	desc = "broken Core AI Module: 'Reconfigures the AI's core laws.'"
+
+/obj/machinery/broken/Initialize()
+	..()
+	explosion(loc, 1, 2, 3, 3)
+	return INITIALIZE_HINT_QDEL
+
+/obj/machinery/broken/Destroy()
+	contents.Cut()
+	return ..()
+
+/obj/item/weapon/aiModule/broken/transmitInstructions(mob/living/silicon/ai/target, mob/sender)
+	..()
+	IonStorm(0)
+	explosion(sender.loc, 1, 1, 1, 3)
```

Comment:
sender will be possibly qdel'd by the time drop_from_inventory is called if this gibs the user.